BatteryMeter.c

```
@@ -25,34 +25,142 @@ static const int BatteryMeter_attributes[] = {
+
+   /* stable: power unknown or |power| < 5 W – time estimate would be unreliable */
+   bool isDischarging = havePower && info.powerCurr <= -5.0;
+   bool isCharging = havePower && info.powerCurr >= 5.0;
+
+   /* time estimate in whole minutes; -1 means not available */
+   int timeMinutes = -1;
+   if (isDischarging && isPositive(info.energyCurr)) {
+      /* floor for discharge; powerCurr is negative, use negative scaling when dividing */
+      timeMinutes = (int)floor(info.energyCurr / info.powerCurr * -60.0);
+   } else if (isCharging && 0.95 * info.energyFull > info.energyCurr) {
+      /* ceil for charge */
```

Cppcheck (2.20.0)

[style] The function 'HeaderLayout_getName' is never used.

(unusedFunction)